src/main/java/org/usfirst/frc/team1806/robot/Robot.java

```diff
@@ -243,5 +244,32 @@ private void runTeleOp(){
       m_oi.runCommands();
       allPeriodic();
     }
+
+    /**
+     * Sets the global game piece mode, runs functions associated with mode change on subsystems.
+     * @param mode the wanted mode
+     */
+    public static synchronized void setGamePieceMode(GamePieceMode mode){
+        GamePieceMode = mode;
+
+        switch(mode){
```

Comment:
I think it's much simpler to directly call the goToCargoMode and goToHatchMode on the subsytems they apply to, not through the subsystem manager. If there are really only ~3 subsystems that need the methods, why are they on the interface? I think it's overdesigned in this respect